### PR TITLE
Unified binary -- part 7

### PR DESCRIPTION
**Pull request by liuw**
This is the last batch for supporting a unified binary.

This PR deals with all the issues when I enable kvm and mshv at the same time. At the end it also contains ~a new command line option and~ changes to GitHub actions.

CC @michael2012z @MrXinWang @jinankjain @russell-islam

### CODE REVIEW

File: hypervisor/src/lib.rs [hunk 1]
```diff
@@ -67,14 +67,42 @@ pub enum HypervisorType {
     Mshv,
 }
 
-pub fn new() -> std::result::Result<Arc<dyn Hypervisor>, HypervisorError> {
-    #[cfg(feature = "kvm")]
-    let hv = kvm::KvmHypervisor::new()?;
+#[derive(Debug, Eq, PartialEq)]
+pub enum HypervisorSelection {
+    AUTO,
```
Comment (rbradford) [inline hypervisor/src/lib.rs:72]: This enum looks wrong. Is there a clippy override somewhere in this file or crate?

File: src/main.rs [hunk 2]
```diff
@@ -372,6 +373,14 @@ fn create_app<'a>(
                 .takes_value(true)
                 .possible_values(&["true", "false", "log"])
                 .default_value("true"),
+        )
+        .arg(
```
Comment (rbradford) [inline src/main.rs:377]: Not convinced we need this. Do situations exist where both are available on the host?

File: hypervisor/src/lib.rs (hunk 1 above)
Comment (liuw) [reply hypervisor/src/lib.rs:72]: Do you mean it should be `Auto` instead?

Yes there is an override in the hypervisor crate.

File: src/main.rs (hunk 2 above)
Comment (liuw) [reply src/main.rs:377]: Not right now. If you want auto section at all times I can drop this argument and other code.

File: hypervisor/src/lib.rs (hunk 1 above)
Comment (rbradford) [reply hypervisor/src/lib.rs:72]: I meant the capitalisation. It should be "Auto, Kvm, Mshv" for consistency with the other enums. I thought clippy would be angry with fully capitalised enum variants.

File: src/main.rs (hunk 2 above)
Comment (rbradford) [reply src/main.rs:377]: I generally prefer to keep things simpler and if there is no benefit to adding it right now then I would prefer to drop it. We can always add it later.

Comment (liuw) [reply src/main.rs:377]: Sure. That's fine by me.

**Review by rbradford — APPROVED** [review]

**Pull request merged**